compiler/rustc_resolve/src/late/diagnostics.rs

```diff
@@ -1539,6 +1539,19 @@ impl<'ast, 'ra, 'tcx> LateResolutionVisitor<'_, 'ast, 'ra, 'tcx> {
                     &poly_trait_ref.trait_ref.path.segments[..]
                 {
                     if ident.span == span {
+                        if matches!(
+                            self.resolve_path(
+                                &[Segment::from_ident(*ident)],
+                                Some(TypeNS),
+                                None,
+                                PathSource::Type,
+                            ),
+                            PathResult::Indeterminate | PathResult::Failed { .. }
+                        ) {
```

Comment:
Not sure whether the use of `resolve_path`, in turn requiring a `&mut self` is necessary. Is it possible to use the partial res?

Reply:
yes, it's simpler to use partial res, fixed.